pkg/envoy/server.go

```diff
@@ -414,7 +419,7 @@ func (s *XDSServer) AddMetricsListener(port uint16, wg *completion.WaitGroup) {
 				Address: &envoy_config_core.Address_SocketAddress{
 					SocketAddress: &envoy_config_core.SocketAddress{
 						Protocol:      envoy_config_core.SocketAddress_TCP,
-						Address:       "::",
+						Address:       listenerAddr,
 						Ipv4Compat:    true,
```

Comment:
I suppose having `Ipv4Compat: true` is harmless when listening on an IPv4 is address?

Reply:
Thanks for your time @jrajahalme . I have just checked the related code of envoy about ipv4Compat, and here's the only usage of socket_address.ipv4_compat, that means IPv4Compat only affect for ipv6 connection. Here's the code:

https://github.com/envoyproxy/envoy/blob/main/source/common/network/utility.cc#L137